packages/braid-design-system/src/lib/components/BraidProvider/BraidProvider.tsx

```diff
@@ -82,10 +82,12 @@ export const BraidProvider = ({
 
   useHideFocusRings(!(alreadyInBraidProvider || inTestProvider));
 
+  // Node 21 introduces an empty navigator object, so checks need a little more logic to them
   assert(
-    typeof navigator !== 'object' ||
-      navigator.userAgent.indexOf('jsdom') === -1 ||
-      inTestProvider,
+    inTestProvider ||
+      typeof navigator === 'undefined' ||
+      typeof navigator.userAgent === 'undefined' ||
```

Comment:
Why a typoeof comparison over a value comparison?

```suggestion
      navigator.userAgent === undefined ||
```